js/foam/u2/search/FilterController.js

```diff
@@ -0,0 +1,451 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+CLASS({
+  package: 'foam.u2.search',
+  name: 'FilterController',
+  extends: 'foam.u2.View',
+
+  traits: [
+    'foam.u2.md.MDViewOverridesTrait',
```

Comment:
I'm under the impression that choosing which `View` type to extend was a Bad Thing about U1 that we're glad to be free of, and that `u2.DetailView` is a leaf, not the root of a large subtree.

To that end, I moved this logic into a trait so it could be reused by others who want these overrides, including users' own controllers.